src/common/commonutils/FileUtils.c

```diff
@@ -514,6 +563,288 @@ int CheckFileSystemMountingOption(const char* mountFileName, const char* mountDi
     return status;
 }
 
+static int CopyMountFile(const char* source, const char* target, void* log)
+{
+    FILE* sourceHandle = NULL;
+    FILE* targetHandle = NULL;
+    struct mntent* mountStruct = NULL;
+    int status = 0;
+
+    if ((NULL == source) || (NULL == target))
+    {
+        OsConfigLogError(log, "CopyMountFile called with invalid argument(s)");
+        return EINVAL;
+    }
+
+    if (!FileExists(source))
+    {
+        OsConfigLogInfo(log, "CopyMountFile: file '%s' not found", source);
+        return EINVAL;
+    }
+        
+    if (NULL != (targetHandle = setmntent(target, "w")))
+    {
+        if (NULL != (sourceHandle = setmntent(source, "r")))
+        {
+            while (NULL != (mountStruct = getmntent(sourceHandle)))
+            {
+                if (0 != addmntent(targetHandle, mountStruct))
+                {
+                    if (0 == (status = errno))
+                    {
+                        status = ENOENT;
+                    }
+                    
+                    OsConfigLogError(log, "CopyMountFile ('%s' to '%s'): failed adding '%s %s %s %s %d %d', addmntent() failed with %d", source, target, 
+                        mountStruct->mnt_fsname, mountStruct->mnt_dir, mountStruct->mnt_type, mountStruct->mnt_opts, mountStruct->mnt_freq, mountStruct->mnt_passno, status);
+                    break;
+                }
+            }
+        }
+        else
+        {
+            if (0 == (status = errno))
+            {
+                status = ENOENT;
+            }
+            
+            OsConfigLogError(log, "CopyMountFile: could not open source file '%s', setmntent() failed (%d)", source, status);
+        }
+        
+        fflush(targetHandle);
+        endmntent(targetHandle);

```

Comment:
Same here, is it worth to check error codes?

Reply:
This is similar to fclose. Same reasons. Typically, these are not checked for errors as there is nothing else to follow up after.